### PR TITLE
Add BuildErrorWithContent EventArgs

### PR DESCRIPTION
**Pull request by yuehuang010**
Fixes #

### CODE REVIEW

File: src/Framework/BuildErrorWithContentEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    /// <summary>
+    /// Arguments for error events with additional content
+    /// </summary>
+    // WARNING: marking a type [Serializable] without implementing
+    // ISerializable imposes a serialization contract -- it is a
+    // promise to never change the type's fields i.e. the type is
+    // immutable; adding new fields in the next version of the type
+    // without following certain special FX guidelines, can break both
+    // forward and backward compatibility
+    [Serializable]
```
Comment (JanKrivanek) [inline src/Framework/BuildErrorWithContentEventArgs.cs:21]: nit: we should stop copying over the `SerializableAttribute` and the comment above.
We use custom serialization for `BuildEventArgs`

**Review by JanKrivanek — APPROVED** [review]
Looks good.
Although additional context of the goal here might be useful

File: src/Framework/BuildErrorWithContentEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    public class BuildErrorWithContentEventArgs : BuildErrorEventArgs
+    {
+        public string AdditionalContentType { get; protected set; }
+        public string AdditionalContentText { get; protected set; }
+        public string AdditionalContentSimpleText { get; protected set; }
```
Comment (rainersigwald) [inline src/Framework/BuildErrorWithContentEventArgs.cs:26]: What's the `SimpleText` field here?

**Review by rainersigwald — COMMENTED** [review]
This looks ok in general; do you need a build or experimental insertion to get the C++ side going or are you sharing work early?

@JanKrivanek the context is #8028.

**Comment by yuehuang010** [comment]
Close old PR.

**Pull request closed without merging**